js/script.js

```diff
@@ -848,3 +848,173 @@ var pushmenu = (function(){
+			}
+		}
+		selectTargetHandler(contentToggleItem, targetId);
+	}
+
+	function selectTargetHandler(contentToggleItem, targetId) {
+		for (let toggleBlockId in toggleBlocksIdsToTocsIds[contentToggleItem.id]) {
+			for (let tocid of toggleBlocksIdsToTocsIds[contentToggleItem.id][toggleBlockId]) {
+				let tocItem = $('#' + tocid);
+				if(tocItem && tocItem.parentNode) {
+					if (contentToggleItem.id === targetId) {
+						tocItem.parent().removeClass('hide');
+						continue;
+					}
+					tocItem.parent().addClass('hide');
+				}
+			}
+		}
+
+		$(".tb-content-toggle#" + contentToggleItem.id + " ." + contentToggleItem.id + " > a.content-toggle-button").removeClass("active");
+		$(".tb-content-toggle#" + contentToggleItem.id + " ." + contentToggleItem.id + " > a.content-toggle-button[data-target='#" + targetId + "']").addClass("active");
+		$(".tb-content-toggle#" + contentToggleItem.id +  " > .panel > .panel-collapse").removeClass("show");
+		$(".tb-content-toggle#" + contentToggleItem.id +  " > .panel > .panel-collapse#" + targetId).addClass("show");
+	}
+
+	function replaceHashWithHeading(id) {
+		const headers = $('#' + id).find(":header");
+		const filteredHeaders = $('#' + id).find('p');
+		const siblingParagraphs = $(filteredHeaders).map(function(idx, el) {
+			return $(el).nextAll('p:first');
+		});
+
+		$(siblingParagraphs).each(function() {
+			if ($(this).text().startsWith('##')) {
+				const paragraph = $(this);
+				const headerLevel = paragraph.text().split(' ')[0].length;
+				const newHeader = $("<h" + headerLevel + ">").text(paragraph.text().replace(/^#+/, ''));
+				newHeader.attr("id", newHeader.text().trim().split(' ').join('-').trim().toLowerCase());
+				paragraph.replaceWith(newHeader);
+			}
+		});
+	}
+
+	function initContentToggleHandler(contentToggleItem) {
+		let toggleBlocksIds = [];
+		$('.panel-heading.' + contentToggleItem.id + ' > a').each(function() {
+			toggleBlocksIds.push($(this).attr("data-target").substring(1));
+		});
+		let i = 0;
+		for (let id of toggleBlocksIds) {
+			toggleBlocksIdsToTocsIds[contentToggleItem.id][id] = [];
+			$('#' + id).find(":header").each(function() {
+				let heading = $(this);
+				heading.attr("id", heading.attr("id") + i++);
+				toggleBlocksIdsToTocsIds[contentToggleItem.id][id].push('markdown-toc-' + $(this).attr('id'));
+			});
+			replaceHashWithHeading(id);
+		}
+
+		window.addEventListener('popstate', function() {
+			onPopStateHandler(contentToggleItem);
+		});
+
+		onPopStateHandler(contentToggleItem);
+
+		$('.tb-content-toggle#' + contentToggleItem.id + ' .' + contentToggleItem.id + ' > a.content-toggle-button')
+			.each((idx,element) => parseButtons(element, contentToggleItem));
+
+		const firstId = Object.keys(toggleBlocksIdsToTocsIds[contentToggleItem.id])[0];
+
+		$('.tb-content-toggle#' + contentToggleItem.id + " ." + contentToggleItem.id + " >  a.content-toggle-button").removeClass("active");
```

Comment:
Perhaps it is better not to add the value of the id to the class but to rewrite using the CSS selector?
`$('.tb-content-toggle#' + contentToggleItem.id + " > .panel > .panel-heading > a.content-toggle-button")`